appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanContainer.java

```diff
@@ -748,7 +754,7 @@ private static boolean isNoMethodsListenerInterface(Class<?> messageListenerType
     }
 
     @Override
-    protected EJBContextImpl _constructEJBContextImpl(Object instance) {
+    protected MessageBeanContextImpl _constructEJBContextImpl(Object instance) {
```

Comment:
API change. But this is protected scope and classes are final. Maybe I could do even more changes.